core/src/main/java/bisq/core/api/CorePaymentAccountsService.java

```diff
@@ -49,17 +52,19 @@ public CorePaymentAccountsService(Config config,
         this.user = user;
     }
 
-    public void createPaymentAccount(String accountName, String accountNumber, String fiatCurrencyCode) {
-        // Create and persist a PerfectMoney dummy payment account.  There is no guard
-        // against creating accounts with duplicate names & numbers, only the uuid and
-        // creation date are unique.
-        PaymentMethod dummyPaymentMethod = PaymentMethod.getDummyPaymentMethod(PaymentMethod.PERFECT_MONEY_ID);
-        PaymentAccount paymentAccount = PaymentAccountFactory.getPaymentAccount(dummyPaymentMethod);
-        paymentAccount.init();
-        paymentAccount.setAccountName(accountName);
-        ((PerfectMoneyAccount) paymentAccount).setAccountNr(accountNumber);
-        paymentAccount.setSingleTradeCurrency(new FiatCurrency(fiatCurrencyCode.toUpperCase()));
-        user.addPaymentAccount(paymentAccount);
+    void createPaymentAccount(String paymentMethodId,
+                              String accountName,
+                              String accountNumber,
+                              String currencyCode) {
+
+        PaymentAccount paymentAccount = getNewPaymentAccount(paymentMethodId,
+                accountName,
+                accountNumber,
+                currencyCode);
+
+        // TODO not sure if there is more to do at account creation.
```

Comment:
Yes there is lots of extra stuff, mainly the many different payment methods with their custom requirements. Input validation, currency selection, country selection, etc...